### PR TITLE
Enable homepage bounties search to include filters

### PR DESCRIPTION
**Pull request by AbdulWahab3181**
## Describe your changes

https://www.loom.com/share/86c634ef14864ac9974be46e476e2b79

## Issue ticket number and link

Closes https://github.com/stakwork/sphinx-tribes-frontend/issues/231

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## Checklist before requesting a review
- [x] I have performed a self-review of my code
- [x] I have tested on Chrome and Firefox
- [ ] I have tested on a mobile device
- [x] I have provided a screenshot or recording of changes in my PR if there were updates to the frontend

### CODE REVIEW

**Comment by Vayras** [comment]
@AbdulWahab3181 let's test this once the workflows are completed

**Comment by AbdulWahab3181** [comment]
**Note**

Whoever is going to test this issue, please ensure they have the changes from [this PR](https://github.com/stakwork/sphinx-tribes/pull/1501) in their backend environment.

**Comment by Vayras** [comment]
@AbdulWahab3181 Good job, does work as intended

**Comment by Vayras** [comment]
okay @AbdulWahab3181  can you include tests for the same? I see they are not included in this pr

**Comment by AbdulWahab3181** [comment]
> okay @AbdulWahab3181 can you include tests for the same? I see they are not included in this pr

Added

File: src/people/widgetViews/BountyHeader.tsx [hunk 1]
```diff
@@ -446,7 +451,12 @@ const BountyHeader = ({
                 onChange={onChangeSearch}
                 onKeyUp={(e: any) => {
                   if (e.key === 'Enter' || e.keyCode === 13) {
```
Comment (elraphty) [inline src/people/widgetViews/BountyHeader.tsx:453]: Put this functionality in an external function, and pass the function to onKeyUp

Comment (AbdulWahab3181) [reply src/people/widgetViews/BountyHeader.tsx:457]: @elraphty Addressed

Comment (elraphty) [reply src/people/widgetViews/BountyHeader.tsx:457]: > @elraphty Addressed

@AbdulWahab3181 handleSearch should receive the event parameter

e.g
```
onKeyUp={handleSearch}

const handleSearch = (e) => {
   if(e) {} 
}
```

Comment (AbdulWahab3181) [reply src/people/widgetViews/BountyHeader.tsx:457]: @elraphty I don't think we need to receive the event parameter. In the onChangeSearch method we sets the [searchText](https://github.com/stakwork/sphinx-tribes-frontend/pull/239/files#diff-f8355a876a6e879aa2e9e4210a2771d0d9e55a556bc54ee5bd8ae3a2d1d4d9daL388) state, which is used inside the getPeopleBounties method [here](https://github.com/stakwork/sphinx-tribes-frontend/blob/master/src/store/main.ts#L857). The handleSearch method is also used [here](https://github.com/stakwork/sphinx-tribes-frontend/pull/239/files#diff-f8355a876a6e879aa2e9e4210a2771d0d9e55a556bc54ee5bd8ae3a2d1d4d9daR402), where we have the search text as an empty string.

Comment (elraphty) [reply src/people/widgetViews/BountyHeader.tsx:457]: This works
```sh
const getBounties = () => {
    main.getPeopleBounties({
      page: 1,
      resetPage: true,
      ...checkboxIdToSelectedMap,
      ...checkboxIdToSelectedMapLanguage
    });
  }

  const handleSearch = (e: any) => {
    if (e.key === 'Enter' || e.keyCode === 13) {
      getBounties();
    }
  };

  const clearSearch = () => {
    getBounties()
  }

  let timeoutId;
  const onChangeSearch = (e: any) => {
    ui.setSearchText(e);
    clearTimeout(timeoutId);
    // Set a new timeout to wait for user to pause typing
    timeoutId = setTimeout(() => {
      if (ui.searchText === '') {
        clearSearch();
      }
    }, 1000);
  };

```

Comment (AbdulWahab3181) [reply src/people/widgetViews/BountyHeader.tsx]: Addressed

**Comment by ecurrencyhodler** [comment]
Tested on staging. This looks great. Thanks Abdul. Paid 200k sats! https://community.sphinx.chat/bounty/1477

**Pull request merged**